src/components/AccessibilityPolicyForm/index.jsx

```diff
@@ -0,0 +1,278 @@
+    const validateEmail = this.validateFormContent('email', formEmail);
+    const validateFullName = this.validateFormContent('fullName', formFullName);
+    const validateMessage = this.validateFormContent('message', formMessage);
+    const validContent = validateEmail.isValid &&
+      validateFullName.isValid && validateMessage.isValid;
+
+    this.setState({
+      statusAlertOpen: true,
+      validFormContent: validContent,
+      validationMessages: [
+        validateEmail.validationMessage,
+        validateFullName.validationMessage,
+        validateMessage.validationMessage,
+      ],
+    });
+
+    if (validContent) {
+      this.props.submitAccessibilityForm(formEmail, formFullName, formMessage);
+    }
+  }
+
+  getStatusFields() {
+    const { accessibilityStatus, email } = this.props;
+    const { validFormContent, validationMessages } = this.state;
+    let status = {
+      alertType: 'info',
+      alertDialog: 'Submitting feedback',
+    };
+
+    if (!validFormContent) {
+      status = {
+        alertType: 'danger',
+        alertDialog: (
+          <div>
+            <div>Please fill in all required fields.</div>
+            <br />
+            <div>
+              <ul>
+                {validationMessages.map((error, index) => {
+                  let errorMessage = '';
+                  if (error) {
+                    /* eslint-disable react/no-array-index-key */
+                    errorMessage = <li key={`Error-${index}`}>{error}</li>;
+                  }
+                  return errorMessage;
+                })}
+              </ul>
+            </div>
+          </div>
+        ),
+      };
+    } else if (accessibilityStatus.type ===
+        accessibilityActions.ACCESSIBILITY_FORM_SUBMIT_RATE_LIMIT_FAILURE) {
+      status = {
+        alertType: 'danger',
+        alertDialog: (
+          <div>
+            We are currently experiencing high volume. Try again later today or
+             send an email message to <a href={`mailto:${email}`}>{email}</a>
+          </div>
+        ),
+      };
+    } else if (accessibilityStatus.type ===
+        accessibilityActions.ACCESSIBILITY_FORM_SUBMIT_SUCCESS) {
+      status = {
+        alertType: 'success',
+        alertDialog: (
+          <div>
+            Thank you for contacting edX!
+            <br />
+            <br />
+            Thank you for your feedback regarding the accessibility of Studio. We typically respond
+             within one business day (Monday to Friday, 13:00 to 21:00 UTC).
+          </div>
+        ),
+      };
+    }
+
+    return status;
+  }
+
+  validateFormContent(inputType, inputValue) {
+    let feedback = { isValid: true };
+    if (inputType === 'email') {
+      /* eslint-disable max-len */
+      /* eslint-disable no-useless-escape */
+      const emailRegEx = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
```

Comment:
This regex fails for the given valid email address according to RFC2822
> !#$%&'*+-/=.?^_`{|}~@[IPv6:0123:4567:89AB:CDEF:0123:4567:89AB:CDEF]
> (from [this StackOverflow answer](https://stackoverflow.com/a/297494)).

I know that email address might seem absurd, and this regex is probably good for 99.99999% of all business cases but the point is why not use a library for email validation instead of having to keep (and maintain) this validation regex?

There's [`isemail`](https://www.npmjs.com/package/isemail), which has been downloaded ~4 million times in the last month and has one dependency, `punycode`, which itself has no dependencies.

__`isemail` does validate the previous email address to `true`__
![image](https://user-images.githubusercontent.com/8136030/33718241-ba6d5066-db2a-11e7-961e-9a13f20066a5.png) ([Link](https://runkit.com/embed/4ysueel4j9r8))

Reply:
Using a library for this to prevent any need for future maintainability is a good idea -- I think for this MVP we'll stick with the regex we have currently, but I'll add a TODO to investigate using that package at a future version, thanks!